src/commands/moderation.rs

```diff
@@ -312,35 +318,43 @@ pub async fn mute(
         pending_unmute.abort();
     }
 
-    data.pending_unmutes.insert(
-        member.user.id.0,
-        queue_unmute_member(
-            &ctx.discord().http,
-            &data.database,
-            &member,
-            mute_role_id,
-            mute_duration.num_seconds() as u64,
-        ),
-    );
+    if !is_eternal {
+        data.pending_unmutes.insert(
+            member.user.id.0,
+            queue_unmute_member(
+                &ctx.discord().http,
+                &data.database,
+                &member,
+                mute_role_id,
+                mute_duration.num_seconds() as u64,
+            ),
+        );
+    }
 
     if result.is_none() {
         if let Err(e) = member.disconnect_from_voice(&ctx.discord().http).await {
             warn!("Could not disconnect member from voice channel: {}", e);
         }
     }
 
+    let duration = if is_eternal {
+        String::new()
+    } else {
+        format!("<t:{}:F>", unmute_time)
+    };
+
     respond_moderation(
         &ctx,
         &ModerationKind::Mute(
             member.user,
             author.clone(),
             reason,
-            format!("<t:{}:F>", unmute_time.timestamp()),
+            duration,
```

Comment:
why is `duration` not an `Option<String>`? Creating an empty string like this is not idiomatic, we have `Option`s for that.

Reply:
https://github.com/revanced/revanced-discord-bot/pull/44#discussion_r1044776789

When duration is empty, it doesn't append the field.

Reply:
This has been resolved